server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -507,6 +509,9 @@ protected void closeInternal() {
         this.compositeIndexSettings = compositeIndexSettings;
         this.fileCache = fileCache;
         this.replicator = replicator;
+        this.cachingEnabledForAllQueries = ENABLE_FOR_ALL_REQUESTS_SETTING.get(clusterService.getSettings());
```

Comment:
[NIT] requestCachingEnabledForAllQueries? As there are other types of caches as well here.